Lib/shutil.py

```diff
@@ -201,7 +205,13 @@ def copyfileobj(fsrc, fdst, length=COPY_BUFSIZE):
 
 def _samefile(src, dst):
     # Macintosh, Unix.
-    if hasattr(os.path, 'samefile'):
+    if isinstance(src, os.DirEntry) and _HAS_SAMESTAT:
+        try:
+            return os.path.samestat(src.stat(), os.stat(dst))
```

Comment:
See https://bugs.python.org/issue30480 .

Not always `samefile()` can be replaced by `samestat()` on Windows.